File: typescript/Creational/02-AbstractFactory/Concrete Factories/HomeownerNotificationFactory.ts
class HomeownerNotificationFactory implements IClientNotificationFactory {
  createAlertNotifier():   IAlertNotifier   { return new SmsStyleAlert(); }
  createEmailReceipt():    IEmailReceipt    { return new SimpleEmailReceipt(); }
  createDashboardWidget(): IDashboardWidget { return new MobileStatusCard(); }
}

File: csharp/Creational/02-AbstractFactory/Concrete Factories/HomeownerNotificationFactory.cs
public class HomeownerNotificationFactory : IClientNotificationFactory
{
    public IAlertNotifier   CreateAlertNotifier()   => new SmsStyleAlert();
    public IEmailReceipt    CreateEmailReceipt()    => new SimpleEmailReceipt();
    public IDashboardWidget CreateDashboardWidget() => new MobileStatusCard();
}